### DISCUSSION TITLE
Setup Dev Env

### DISCUSSION THREAD

**Discussion by aantipov** [DamianOsipiuk/vue-query#104] [Q&A]
Hi, 

Thanks a lot for making the library.
I'd like to play with its API - try some changes and see how they work. 
And, probably, contribute to the Library.

I'm trying to setup development environment and I'm experiencing some problems.

My steps:
1. Forked and cloned `vue-query`
2. run 
- `npm install`
- `npm link`
- `npm publish`
3. In my project folder I run
- `npm link vue-query`
- `npm run dev` (using Vite)

When I open a page, I see the following in the Console:
![image](https://user-images.githubusercontent.com/427621/147885258-68d9bbea-deb3-412a-bd6b-dea3de999e28.png)

Normally, I don't have any warnings, everything works fine. But with the "linked" version of the package I have the problems as in the picture.

Do you know what could go wrong?
What is you dev setup?

Node/Npm versions
![image](https://user-images.githubusercontent.com/427621/147885343-9cc76b86-1555-4980-afe3-4a5db345f624.png)

--
Alexey

**Comment by aantipov** [comment]
I found a way to fix the problem - install `vue-demi` package in my project folder and adjust Vite config 
![image](https://user-images.githubusercontent.com/427621/148105835-bdb6a036-e892-4c06-8cd2-2c3f094bcc10.png)

Though I don't really need `vue-demi` in my project. But for now, I don't know better way

**Reply by DamianOsipiuk** [reply]
Hi,
I think vue-demi should be installed as a dependency of vue-query anyway, so you do not have to install it manually.
However vue-demi has a postinstall script to detect installed vue version and change entry file. So if you have some security measures to disable postinstall scripts it might not work out of the box.
As for vite i think you need to disable vue-query optimization as well, so you would not need to remove cache folder after rebuilding changes every time.

**Reply by aantipov** [reply]
There is no problem when using Vue-query normally.
I have the problem only when I `npm link` it. Changing the Vite's config is also required.

My guess was that I'm using the "raw" library, not the compiled one. But that's not true. Changes to it are reflected in my app only after building the library.

I don't really know what's going on. Also I could not make DevTools working with the linked library.

Anyway, I found a workaround. So I'm closing the discussion for now.
